portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -427,20 +432,25 @@ else if (path.startsWith("/user_portrait")) {
 		return null;
 	}
 
-	protected FileEntry getFileEntry(String[] pathArray) throws Exception {
+	protected FileEntry getFileEntry(
+			String[] pathArray, HttpServletRequest httpServletRequest)
+		throws Exception {
+
+		FileEntry fileEntry;
```

Comment:
Can you move the logic at the end of the method to a `_checkExpiredFileEntry(FileEntry fileEntry)` method?

Also, avoid local variables with such a large extent. Each branch should look something like:
```
FileEntry fileEntry = DLAppServiceUtil.getFileEntry(...);

_checkExpiredFileEntry(fileEntry);

return fileEntry;
```